qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -0,0 +1,293 @@
+    expanded in a truncated set of orbitals that define an active space,
+
+    .. math::
+
+        \hat A = \sum_{\alpha \leq 2N_\mathrm{docc}} \langle \alpha \vert \hat{\mathcal{A}}
+        \vert \alpha \rangle ~ \hat{n}_\alpha +
+        \sum_{\alpha, \beta ~ \in ~ \mathrm{active~space}} \langle \alpha \vert \hat{\mathcal{A}}
+        \vert \beta \rangle ~ \hat{c}_\alpha^\dagger\hat{c}_\beta,
+
+        \hat B = \frac{1}{2} \left\{ \sum_{\alpha, \beta \leq 2N_\mathrm{docc}}
+        \langle \alpha, \beta \vert \hat{\mathcal{B}} \vert \beta, \alpha \rangle
+        ~ \hat{n}_\alpha \hat{n}_\beta + \sum_{\alpha, \beta, \gamma, \delta ~
+        \in ~ \mathrm{active~space}} \langle \alpha, \beta \vert \hat{\mathcal{B}}
+        \vert \gamma, \delta \rangle ~ \hat{c}_{\alpha}^\dagger \hat{c}_{\beta}^\dagger
+        \hat{c}_{\gamma} \hat{c}_{\delta} \right\}.
+
+    In the latter equations :math:`N_\mathrm{docc}` denotes the doubly-occupied orbitals,
+    if any, not included in the active space and
+    :math:`\langle \alpha \vert \hat{\mathcal{A}} \vert \beta \rangle` and
+    :math:`\langle \alpha, \beta \vert\hat{\mathcal{B}} \vert \gamma, \delta \rangle`
+    are the matrix elements of the one- and two-particle operators
+    :math:`\hat{\mathcal{A}}` and :math:`\hat{\mathcal{B}}`, respectively.
+
+    The function utilizes tools of `OpenFermion <https://github.com/quantumlib/OpenFermion>`_
+    to buil the second-quantized operator and map it to basis of Pauli matrices via the
+    Jordan-Wigner or Bravyi-Kitaev transformation. Finally, the qubit operator is
+    converted to a a PennyLane observable by the function :func:`~.convert_observable`.
+
+    **Example**
+
+    >>> s2_me_table, init_term = get_s2_me('h2', './pyscf/sto-3g')
+    >>> s2_obs = observable(s2_me_table, init_term=init_term)
+    >>> print(type(s2_obs))
+    <class 'pennylane.vqe.vqe.Hamiltonian'>
+
+    >>> print(s2_obs)
+    (0.75) [I<Wires = [0]>]
+    + (0.375) [Z<Wires = [1]>]
+    + (-0.375) [Z<Wires = [0]> Z<Wires = [1]>]
+    + (0.125) [Z<Wires = [0]> Z<Wires = [2]>]
+    + (0.375) [Z<Wires = [0]>]
+    + (-0.125) [Z<Wires = [0]> Z<Wires = [3]>]
+    + (-0.125) [Z<Wires = [1]> Z<Wires = [2]>]
+    + (0.125) [Z<Wires = [1]> Z<Wires = [3]>]
+    + (0.375) [Z<Wires = [2]>]
+    + (0.375) [Z<Wires = [3]>]
+    + (-0.375) [Z<Wires = [2]> Z<Wires = [3]>]
+    + (0.125) [Y<Wires = [0]> X<Wires = [1]> Y<Wires = [2]> X<Wires = [3]>]
+    + (0.125) [Y<Wires = [0]> Y<Wires = [1]> X<Wires = [2]> X<Wires = [3]>]
+    + (0.125) [Y<Wires = [0]> Y<Wires = [1]> Y<Wires = [2]> Y<Wires = [3]>]
+    + (-0.125) [Y<Wires = [0]> X<Wires = [1]> X<Wires = [2]> Y<Wires = [3]>]
+    + (-0.125) [X<Wires = [0]> Y<Wires = [1]> Y<Wires = [2]> X<Wires = [3]>]
+    + (0.125) [X<Wires = [0]> X<Wires = [1]> X<Wires = [2]> X<Wires = [3]>]
+    + (0.125) [X<Wires = [0]> X<Wires = [1]> Y<Wires = [2]> Y<Wires = [3]>]
+    + (0.125) [X<Wires = [0]> Y<Wires = [1]> X<Wires = [2]> Y<Wires = [3]>]
```

Comment:
Nice example! Quick comment though:

* @mariaschuld has temporarily removed the Wires class from Observables, so the above example should be changed to `(0.125) [X0 Y1 X2 Y3]` etc.

* Even when the Wires class is re-introduced, the above is a bug --- we don't want the output to look like that! So the `Hamiltonian` class will be modified to make sure it continues to print as before.

Reply:
Nice!

Reply:
Done